dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/dataset/DatasetVersion.java

```diff
@@ -472,799 +425,235 @@ public boolean isDeaccessioned() {
             } else {
-                List<DataFile> current = new ArrayList<>();
-                List<DataFile> previous = new ArrayList<>();
-                for (FileMetadata fmdc : this.getFileMetadatas()) {
-                    current.add(fmdc.getDataFile());
-                }
-                for (FileMetadata fmdc : this.getDataset().getReleasedVersion().getFileMetadatas()) {
-                    previous.add(fmdc.getDataFile());
-                }
-                for (DataFile fmd : current) {
-                    previous.remove(fmd);
-                }
+                List<DataFile> current = fileMetadatas.stream()
+                        .map(FileMetadata::getDataFile)
+                        .collect(Collectors.toList());
+                List<DataFile> previous = dataset.getReleasedVersion().getFileMetadatas().stream()
+                        .map(FileMetadata::getDataFile)
+                        .collect(Collectors.toList());
+                previous.removeAll(current);
                 return previous.isEmpty();
             }
         }
         return true;
     }
 
     public boolean isHasPackageFile() {
-        if (this.fileMetadatas.isEmpty()) {
+        if (fileMetadatas.size() != 1) {
             return false;
         }
-        if (this.fileMetadatas.size() > 1) {
-            return false;
-        }
-        return this.fileMetadatas.get(0).getDataFile().getContentType().equals(PackageMimeType.DATAVERSE_PACKAGE.getMimeValue());
+        return PackageMimeType.DATAVERSE_PACKAGE.getMimeValue().equals(fileMetadatas.get(0).getDataFile().getContentType());
     }
 
+    // XHTML
     public boolean isHasNonPackageFile() {
-        if (this.fileMetadatas.isEmpty()) {
-            return false;
-        }
         // The presence of any non-package file means that HTTP Upload was used (no mixing allowed) so we just check the first file.
-        return !this.fileMetadatas.get(0).getDataFile().getContentType().equals(PackageMimeType.DATAVERSE_PACKAGE.getMimeValue());
+        return !fileMetadatas.isEmpty()
+                && !PackageMimeType.DATAVERSE_PACKAGE.getMimeValue().equals(fileMetadatas.get(0).getDataFile().getContentType());
     }
 
     public DatasetVersion cloneDatasetVersion() {
-        DatasetVersion dsv = new DatasetVersion();
-        dsv.setVersionState(this.getPriorVersionState());
-        dsv.setFileMetadatas(new ArrayList<>());
+        DatasetVersion cloned = new DatasetVersion();
+        cloned.setVersionState(getPriorVersionState());
+        cloned.setFileMetadatas(new ArrayList<>());
 
-        if (this.getUNF() != null) {
-            dsv.setUNF(this.getUNF());
+        if (UNF != null) {
+            cloned.setUNF(UNF);
         }
-
-        if (this.getDatasetFields() != null && !this.getDatasetFields().isEmpty()) {
-            dsv.setDatasetFields(DatasetFieldUtil.copyDatasetFields(this.getDatasetFields()));
+        if (datasetFields != null && !datasetFields.isEmpty()) {
```

Comment:
Nearly the same: in case of null value the `for` in `copyDatasetFields` will produce NPE, however this is highly impobable as `datasetFields` is initialized to empty list, so I've removed that `if`.